### PR TITLE
If approved 06-PATCH_/api/article/article_id complete

### PR DESCRIPTION
**Pull request by yahya003**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -117,4 +117,53 @@ describe("GET: /api/users", () => {
...
+  const newVote = {
+    inc_vote: 3
+  }
+  return request(app)
+    .patch("/api/articles/1000")
+    .send(newVote)
+    .expect(404)
+ })
+
+ test("500: responds with server error if the newVote does not contain the key inc_vote", () => {
```
Comment (itskatherine) [inline __tests__/app.test.js:160]: This shouldn't be an internal server error, this is a bad request so would be a 400 error

File: __tests__/app.test.js [hunk 2]
```diff
@@ -117,4 +117,53 @@ describe("GET: /api/users", () => {
       .expect(404)
     })
 })
-      
\ No newline at end of file
+
+describe('6. PATCH /api/articles/:article_id  ', () => {
+test("200: responds with updated votes", () => {
+  const newVote = {
+    inc_vote: 3
```
Comment (itskatherine) [inline __tests__/app.test.js:124]: A small note but the ticket specified the key should be called inc_vote**s** not inc_vote

File: __tests__/app.test.js [hunk 3]
```diff
@@ -117,4 +117,53 @@ describe("GET: /api/users", () => {
...
+ test("500: responds with server error if the newVote does not contain the key inc_vote", () => {
+  const newVote = {
+    vote: 3
+  }
+  return request(app)
+    .patch("/api/articles/6")
+    .send(newVote)
+    .expect(500)
+ })
+})
```
Comment (itskatherine) [inline __tests__/app.test.js:169]: You could also add a test to check if a negative number on the key of inc_votes will be processed correctly (like the example on the ticket)

File: __tests__/app.test.js [hunk 4]
```diff
@@ -117,4 +117,53 @@ describe("GET: /api/users", () => {
...
+ })
+
+ test("400: responds with invalid input for a bad request", () => {
+  const newVote = {
+    inc_vote: 3
+  }
+  return request(app)
+    .patch("/api/articles/%")
+    .send(newVote)
+    .expect(400)
```
Comment (itskatherine) [inline __tests__/app.test.js:147]: Should also check for a return message like 'bad request'

File: __tests__/app.test.js [hunk 5]
```diff
@@ -117,4 +117,53 @@ describe("GET: /api/users", () => {
...
+ })
+
+ test("404: responds with not found error if the article doesn't exist but the entry is valid", () => {
+  const newVote = {
+    inc_vote: 3
+  }
+  return request(app)
+    .patch("/api/articles/1000")
+    .send(newVote)
+    .expect(404)
```
Comment (itskatherine) [inline __tests__/app.test.js:157]: Also please check for a returned error message here to describe what's gone wrong

File: __tests__/app.test.js (hunk 1 above)
Comment (itskatherine) [inline __tests__/app.test.js:160]: We generally don't need to check for 500 server errors, as they should never be happening, we should always catch them before they get to this point. 
You should instead test for a request body that does not have the key of inc_votes. In this case I would still expect a 200 (we don't want a key to be compulsory as we may chose to extend the functionality of the endpoint at a later date.) You can return the review unchanged but you will need inc_votes to have a default value in your code.

File: controllers/controller.js [hunk 6]
```diff
@@ -23,4 +23,14 @@ exports.getUsers = (req, res, next) => {
         res.status(200).send({ users });
   })
   .catch(next)
+};
+
+exports.patchArticleByID = (req, res, next) => {
+  const {article_id} = req.params
```
Comment (itskatherine) [inline controllers/controller.js:29]: Nice destructuring

**Review by itskatherine — COMMENTED** [review]
Good stuff so far Yahya, please look over my comments and address the issues, could you please DM me for another review of this pull request once you've added! 🙂
